### PR TITLE
graphQl api testing setup

### PR DESCRIPTION
**Pull request by harshp92**
1. Graph Ql queries and API test setup
2. Folder structure for the e2e test repo

### CODE REVIEW

**Review by raja-huspy — APPROVED** [review]
Looks fine  & Approved

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
-
+.idea
+package-lock.json
+node_modules
+videos
```
Comment (alexAtHuspy) [inline .gitignore:4]: do you want to add an empty line to the end of the file

File: cypress/e2e/propertyListing.cy.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+let propertyListings = new PropertyListing();
+describe('Property Listing', () => {
+  it('Validate spain property listing', () => {
+    propertyListings.getPropertyList('SPAIN', 'FEATURED')
+    cy.get('@propertyList').then((array) => 
+    array.forEach(element => {
+      expect(element.title).not.be.empty;
+    }));
+  })
+})
```
Comment (alexAtHuspy) [inline cypress/e2e/propertyListing.cy.js:12]: consider adding an empty line to the end of the file

File: cypress/support/e2e.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (alexAtHuspy) [inline cypress/support/e2e.js:20]: need an empty line

File: cypress/support/graphql-queries.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export function createPropertyListingQuery(country, sortCriteria){
+    return `query{publishedProperties(input: {country:${country} sort:${sortCriteria}}first:20){nodes{title price} pageInfo{startCursor endCursor}}}`;
+
```
Comment (alexAtHuspy) [inline cypress/support/graphql-queries.js:3]: remove empty line 3 and add an empty line to the end of the file

**Review by alexAtHuspy — APPROVED** [review]

File: cypress/moduleObjects/propertyListings/propertyList.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { createPropertyListingQuery } from "../../support/graphql-queries";
+
+export class PropertyListing{
```
Comment (AlexeyAltunin) [inline cypress/moduleObjects/propertyListings/propertyList.js:3]: let's use the same class and file name. PropertyListing or PropertyList

File: package.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "api-automation",
+  "version": "1.0.0",
+  "description": "E2E Cypress Api Automation",
+  "main": "index.js",
+  "scripts": {
```
Comment (AlexeyAltunin) [inline package.json:6]: can we add lint/prettier to keep code similar everywhere (can be taken from real-estate-fe or mobile app

**Review by AlexeyAltunin — CHANGES_REQUESTED** [review]

**Pull request merged**